### ISSUE TITLE
4XX, 5XX... standard return codes

### ISSUE THREAD

**Issue by od-forsk** [baloise-incubator/spectral-ruleset#38]
Hello,

Return code 1XX ... 5XX are defined in the OpenAPI spec: https://swagger.io/specification/#patterned-fields-1
Could you please add them to the rules:
- must-use-standard-http-status-codes
- should-use-well-understood-http-status-codes

`should-use-well-understood-http-status-codes` especially is difficult to override as it uses a custom function.

**Comment by dweber019** [comment]
Hi @od-forsk 

I think we could use https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml as the source for must-use-standard-http-status-codes. Do you agree?

Regarding should-use-well-understood-http-status-codes, could you propose a solution for implementing this check differently?

**Comment by nmoreaud** [comment]
Hello @dweber019. do you think `5XX` should be added as a valid http code?
Its usage is described on many documentation pages, such as https://swagger.io/docs/specification/describing-responses/
Ex:
```yaml
      responses:
        '200':
          description: OK
        '400':
          description: Bad request. User ID must be an integer and larger than 0.
        '401':
          description: Authorization information is missing or invalid.
        '404':
          description: A user with the specified ID was not found.
        '5XX':
          description: Unexpected error.
```

I don't know if it is comparable to the "default" response item: https://opensource.zalando.com/restful-api-guidelines/#151

**Comment by dweber019** [comment]
First I like to say that our intention is to stay compliant to the Zalando guide. This said, I think that the `default` response can handle "401 (unauthenticated), 403 (unauthorized) or 404 (not found)" as they are straightforward.
For other error codes I would suggest to define them, but according to

> However, you should not use it and explicitly define the error code as soon as it provides endpoint specific indications for clients of how to avoid calling the endpoint in the wrong way, or be prepared to react on specific error situation. https://opensource.zalando.com/restful-api-guidelines/#151

It's more like a `should`.

We could allow using `1xx` to `5xx` for rule `must-use-standard-http-status-codes`, but this would be wrong as an invalide code like `105` would not fail the linter.
Therefor I suggest to overwrite the rule `must-use-standard-http-status-codes`, if you like to have `1xx` to `5xx` accepted.

What do you think?

**Comment by nmoreaud** [comment]
We though more about 4XX and 5XX which can behave like "default" but have a slightly more specific description.
But I'm okay with you, we can close this ticket.
